app/javascript/lib/validation/validate-file-content.js

```diff
@@ -280,17 +392,22 @@ function validateClusterCoordinates(parsedHeaders) {
 }
 
 /** parse a metadata file, and return an array of issues, along with file parsing info */
-export async function parseMetadataFile(chunker, mimeType) {
+export async function parseMetadataFile(chunker, mimeType, fileOptions) {
   const { parsedHeaders, delimiter } = await getParsedHeaderLines(chunker, mimeType, 2)
 
   let issues = validateCapFormat(parsedHeaders, delimiter)
   issues = issues.concat(validateNoMetadataCoordinates(parsedHeaders))
+  if (fileOptions.use_metadata_convention) {
+    issues = issues.concat(validateRequiredMetadataColumns(parsedHeaders))
+  }
   // add other header validations here
 
   const dataObj = {} // object to track multi-line validation concerns
   await chunker.iterateLines((line, lineNum, isLastLine) => {
     const parsedLine = parseLine(line, delimiter)
     issues = issues.concat(validateUniqueCellNamesWithinFile(parsedLine, isLastLine, dataObj))
+    issues = issues.concat(validateMetadataLabelMatches(parsedHeaders, parsedLine, isLastLine, dataObj))
+    issues = issues.concat(validateGroupColumnCounts(parsedHeaders, parsedLine, isLastLine, dataObj))
```

Comment:
For readability, if/when those suggested changes in the upstream PR are done and merged here:

```suggestion
    issues = issues.concat(validateMetadataLabelMatches(headers, line, isLastLine, dataObj))
    issues = issues.concat(validateGroupColumnCounts(headers, line, isLastLine, dataObj))
```